File: Codeforces/800-900/StonesontheTable.java
package Codeforces;

/*
 * url: https://vjudge.net/solution/58633195/pqmKS3mMPAp7OEVG2TYH
 * date: 24 - 02 - 2025
 * category: implementation
 * complexity: O(n)
 * appproach: we need to find the minimum number of stones to remove to make any two adjacent stones have different colors. so after abstracting the problem we can define it as we need to check every to consecutive stones and check if they are the same color or not
 * 
 * instead of getting our hands dirty with indexes, i used a stack to keep track of how many stones we need to remove, then subtract the total number of stones from the size of the stack
 */
import java.util.*;

public class StonesontheTable {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = Integer.parseInt(sc.nextLine());
    String s = sc.nextLine();
    Stack<Character> stk = new Stack<>();
    stk.add(s.charAt(0));

    for (int i = 1; i < n; i++) {
      if (stk.peek() != s.charAt(i)) {
        stk.add(s.charAt(i));
      }
    }

    System.out.println(n - stk.size());

    sc.close();
  }
}

File: Codeforces/800-900/StonesontheTable.cpp
/*
 * url: https://vjudge.net/solution/58924344/NbZk7WnCyi0Miu3vgvlo
 * date: 06 - 03 - 2025
 * category: implementation
 * complexity: O(n)
 */

#include <bits/stdc++.h>
using namespace std;

int main()
{
  int n{};
  cin >> n;
  string str{};
  cin >> str;

  int cnt{};
  for (int i = 1; i < str.length(); i++)
    if (str[i] == str[i - 1])
      cnt++;

  cout << cnt << "\n";
  return 0;
}